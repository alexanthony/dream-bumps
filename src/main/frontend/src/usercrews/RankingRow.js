// @flow
import React, {PropTypes, Component} from 'react';
import { TableRow, TableRowColumn } from 'material-ui/Table';
import { connect } from 'react-redux';
import { Link } from 'react-router';
import FlatButton from 'material-ui/FlatButton';

import { getUser } from '../rootReducer';
import { getCollegeName } from '../utils/colleges';

class RankingRow extends Component {
  render() {
    if (!this.props.ranking) {
      return <TableRow>Loading...</TableRow>
    }
    return (
      <TableRow>
        <TableRowColumn style={{paddingLeft:0, paddingRight:0, width: '45px'}}>{this.props.number}</TableRowColumn>
        {this.props.user ?
          <TableRowColumn>
            <FlatButton
              containerElement={<Link to={`/rankings/${this.props.user.id}`}/>}
              label={`${this.props.user.firstName} ${this.props.user.lastName}`}
              labelStyle={{textTransform: '', fontSize: '13px'}}
              style={{paddingLeft: 0, textAlign: '', marginLeft: '-16px'}}
            />
          </TableRowColumn>
        :
          <TableRowColumn>
            <FlatButton
              containerElement={<Link to={`/rankings/${this.props.ranking.userId}`}/>}
              label={`User ${this.props.ranking.userId}`}
              labelStyle={{textTransform: '', fontSize: '13px'}}
              style={{paddingLeft: 0, textAlign: '', marginLeft: '-16px'}}
            />
          </TableRowColumn>
        }
        {this.props.user ?
          <TableRowColumn>{getCollegeName(this.props.user.college)}</TableRowColumn>
        :
          <TableRowColumn />
        }
        <TableRowColumn style={{paddingRight:0, width: '88px'}}>{this.props.ranking.bumps}</TableRowColumn>
        <TableRowColumn style={{paddingRight:0, width: '88px'}}>{this.props.ranking.cash.toFixed(2)}</TableRowColumn>
        <TableRowColumn style={{paddingRight:0, width: '88px'}}>{this.props.ranking.value.toFixed(2)}</TableRowColumn>
        <TableRowColumn style={{paddingRight:0, width: '88px'}}>{this.props.ranking.dividends.toFixed(2)}</TableRowColumn>
        <TableRowColumn style={{paddingRight:0, width: '88px'}}>{(this.props.ranking.cash + this.props.ranking.value - (this.props.ranking.sex === 'combined' ? 2000 : 1000) - this.props.ranking.dividends).toFixed(2)}</TableRowColumn>
        <TableRowColumn style={{paddingRight:0, width: '88px'}}>{(this.props.ranking.cash + this.props.ranking.value).toFixed(2)}</TableRowColumn>
      </TableRow>
    );
  }
}

RankingRow.propTypes = {
  number: PropTypes.number,
  ranking: PropTypes.shape({
    userId: PropTypes.number,
    cash: PropTypes.number,
    value: PropTypes.number,
    bumps: PropTypes.number,
    dividends: PropTypes.number,
    sex: PropTypes.string
  }),
  user: PropTypes.shape({
    college: PropTypes.string,
    firstName: PropTypes.string,
    lastName: PropTypes.string,
    id: PropTypes.number
  })
}

export const mapStateToProps = (state:Object, ownProps:{ranking:{userId:number}}) => {
  if (ownProps.ranking) {
    return {user: getUser(state)(ownProps.ranking.userId)}
  }
  return {user: {}};
}

export {RankingRow};

export default connect(mapStateToProps)(RankingRow);
