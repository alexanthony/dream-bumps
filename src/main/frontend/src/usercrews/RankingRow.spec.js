import React from 'react';
import { shallow } from 'enzyme';
import { RankingRow } from './RankingRow';
import toJson from 'enzyme-to-json';

test('Renders properly', () => {
  const ranking = {
    userId: 5,
    cash: 11.01,
    value: 1036.5,
    bumps: 3,
    dividends: 12.67,
    sex: 'male'
  };
  const user = {
    id: 5,
    college: null,
    firstName: 'ThisIsAUserName',
    lastName: ''
  };
  const component = shallow(
    <RankingRow
      number={3}
      ranking={ranking}
      user={user}
    />
  );
  expect(component).toBeDefined();
  expect(toJson(component)).toMatchSnapshot()
});
