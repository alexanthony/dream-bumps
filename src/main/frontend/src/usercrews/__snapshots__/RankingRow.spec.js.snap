exports[`test Renders properly 1`] = `
<TableRow
  displayBorder={true}
  hoverable={false}
  hovered={false}
  selectable={true}
  selected={false}
  striped={false}>
  <TableRowColumn
    hoverable={false}
    style={
      Object {
        "paddingLeft": 0,
        "paddingRight": 0,
        "width": "45px",
      }
    }>
    3
  </TableRowColumn>
  <TableRowColumn
    hoverable={false}>
    <FlatButton
      containerElement={
        <Link
          onlyActiveOnIndex={false}
          style={Object {}}
          to="/rankings/5" />
      }
      disabled={false}
      label="ThisIsAUserName "
      labelPosition="after"
      labelStyle={
        Object {
          "fontSize": "13px",
          "textTransform": "",
        }
      }
      onKeyboardFocus={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onTouchStart={[Function]}
      primary={false}
      secondary={false}
      style={
        Object {
          "marginLeft": "-16px",
          "paddingLeft": 0,
          "textAlign": "",
        }
      } />
  </TableRowColumn>
  <TableRowColumn
    hoverable={false} />
  <TableRowColumn
    hoverable={false}
    style={
      Object {
        "paddingRight": 0,
        "width": "88px",
      }
    }>
    3
  </TableRowColumn>
  <TableRowColumn
    hoverable={false}
    style={
      Object {
        "paddingRight": 0,
        "width": "88px",
      }
    }>
    11.01
  </TableRowColumn>
  <TableRowColumn
    hoverable={false}
    style={
      Object {
        "paddingRight": 0,
        "width": "88px",
      }
    }>
    1036.50
  </TableRowColumn>
  <TableRowColumn
    hoverable={false}
    style={
      Object {
        "paddingRight": 0,
        "width": "88px",
      }
    }>
    12.67
  </TableRowColumn>
  <TableRowColumn
    hoverable={false}
    style={
      Object {
        "paddingRight": 0,
        "width": "88px",
      }
    }>
    34.84
  </TableRowColumn>
  <TableRowColumn
    hoverable={false}
    style={
      Object {
        "paddingRight": 0,
        "width": "88px",
      }
    }>
    1047.51
  </TableRowColumn>
</TableRow>
`;
